pyiron/vasp/potential.py

```diff
@@ -272,3 +272,41 @@ def find_potential_file(file_name=None, xc=None, path=None, pot_path_dict=None):
                     if file_name in file_lst:
                         return os.path.join(path, file_name)
     raise ValueError("Either the filename or the functional has to be defined.")
+
+
+def get_enmax_among_species(symbol_lst, return_list=False, xc="PBE"):
+    """
+    Given a list of species symbols, finds the largest applicable encut.
+
+    Args:
+        symbol_lst (list): The list of species symbols.
+        return_list (bool): Whether to return the list of all ENMAX values (in the same order as `species_lst` along with
+            the largest value). (Default is False.)
+        xc ("GGA"/"PBE"/"LDA"): The exchange correlation functional for which the POTCARs were generated. (Default is "PBE".)
+
+    Returns:
+        (float): The largest ENMAX among the POTCAR files for all the species.
+        [optional](list): The ENMAX value corresponding to each species.
+    """
+    from pyiron.vasp.base import Potcar
```

Comment:
We should do the import at the top of the file - that is recommended by PEP8. The only point where we import in the function is when it is a dependency which might not be available on all systems, like `ovito`.

Reply:
No can do (trace below). This introduces a cyclic dependency between potential.py and base.py. 

I think the real solution is to change the architecture. There are a few choices:
- `get_enmax_among_species` could live directly in in base.py. I don't like this -- potential stuff should be in the potential module.
- `get_enmax_among_species` could not be a method of the vasp job type. I don't like this -- it needs to be tab-completable from somewhere the user will have in her notebook.
- `pot_path_dict` could be moved from belonging to the `Potcar` class to somewhere else, e.g. inside the potential module -- this makes a lot of sense to me.
- The `Potcar` class could be moved into the potential module -- this makes some sense to me, but then `Incar`, `Kpoints`, etc would need to be moved out of base for consistency.

I'm perfectly happy to bend a little Pep8 to delay resolving this issue.

Reply:
@sudarsan-surendralal Can we move the `POTCAR` class to a separate file? That should resolve this issue. https://github.com/pyiron/pyiron/blob/enmax_finder/pyiron/vasp/base.py#L2184

Reply:
@jan-janssen @sudarsan-surendralal if we're going to move it, is there a reason not to simply move it into potential.py?

Reply:
I am also fine with that.

Reply:
Yes moving it to `potential.py` is a good idea

Reply:
Ok, I refactored it over. All the vasp tests run fine on Garching; we'll see how it turns out on remote.

Reply:
If all the tests pass, I guess we're all satisfied and can merge it in.